app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
So basically, whether you choose to convert the offers from a JS object to an Immutable object or not, you need to make sure that the empty offers list is consistent with the object type that you choose, so that the empty list is handled the same way as the non-empty list. In this case, that would just mean making sure that the empty list is an empty JS object (note: *object*, not array, since your offers list is actually an object and not an array).